core/src/main/kotlin/ch/ergon/dope/resolvable/fromable/Bucket.kt

```diff
@@ -17,3 +18,11 @@ class AliasedBucket(name: String, val alias: String) : Bucket(name) {
 }
 
 fun AliasedBucket.field(field: String): Field<StringType> = Field(field, alias)
+
+fun Bucket.all(): Expression = Field<StringType>(
+    "*",
+    when (this) {
+        is AliasedBucket -> alias
+        else -> name
```

Comment:
I prefer it when you write `is UnaliasedBucket -> ` instead of the else-branch. Then we see directly that there are only two Buckets.
For that you have to make the bucket _sealed_ instead of _abstract_. In kotlin normally you never use abstract classes, please makt the JoinOperator _sealed_ too

Reply:
done